packages/image_picker/image_picker_platform_interface/lib/src/types/media_selection_options.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:image_picker_platform_interface/image_picker_platform_interface.dart';
+
+/// Options class for [ImagePickerPlatform.getMedia].
+class MediaSelectionOptions {
+  /// Construct a new MediaSelectionOptions instance.
+  MediaSelectionOptions({
+    List<RetrieveType>? types,
```

Comment:
We do we need both an array and a catch-all enum type? It seems like this allows any of `[media]`, `[images, video]`, `[images, media]`, `[video, media]`, or `[images, video, media]` to express the same thing, which is confusing.

Reply:
(Or `[]` or `null`, in fact. We should also make this non-nullable to avoid having two ways for it to be empty.)